src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1607,5 +1611,7 @@ public void ForkChoiceUpdated(Hash256? finalizedBlockHash, Hash256? safeBlockHas
                 _metadataDb.Set(MetadataDbKeys.SafeBlockHash, Rlp.Encode(SafeHash!).Bytes);
             }
         }
+
+        public Block[] GetInvalidBlocks() => _badBlockStore.GetAll();
```

Comment:
I would drop this method from BlockTree, if someone wants to load them, he can go through `BadBlockStore` directly

Reply:
Are you saying to make `_badBlockStore` public?

Reply:
pass BadBlockStore to RPC and get from there, don't get bad blocks through block tree

Reply:
I still can't understand, the block tree is the only place where a reference is stored to the `BlockStore` for bad blocks, how can `DebugBridge` or `DebugRpcModule` get a reference from somewhere else?